src/components/appNavigation/TabsView.js

```diff
@@ -53,13 +48,43 @@ type TabViewProps = {
 //   </View>
 // )
 
-const TabsView = React.memo((props: TabViewProps) => {
-  const { navigation, styles } = props
+const etoroButtons = (goToRewards, styles) => (
```

Comment:
any function rendering jsx is a functional component and should be declared as component:

```
// move styles at the top

// let's unify onPress handlers names
const EToroButtons = withStyles(styles)(({ onPress, styles }) => (
  <>
    // jsx here
  </>
))

const MarketButtons = ({ onPress, style }) => (
  <>
  </>
)


const SupportButton = ({ onPress, style }) => (
 <>
  </>
)

// ...
const marketButtonsStyles = useMemo(() => [
  styles.marketIconBackground, 
  styles.marginRight5
], []);

// ...
{sEToro ? ( 
  <ETtoroButtons onPress={goToRewards} /> 
) : (
  <SupportButton onPress={goToSupport} style={styles.marginLeft10} />
)}
{market && (
  <MarketButtons onPress={goToMarketplace} style={marketButtonsStyles} />
)}
{isEToro && (
  <SupportButton onPress={goToSupport} />
)}
```

Reply:
done